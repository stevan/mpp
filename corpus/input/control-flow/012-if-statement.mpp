# Simple if statement
if ($x > 10) {
    print("big");
}

# If with multiple statements
if ($valid) {
    $count = $count + 1;
    print($count);
    log_message("incremented");
}

# If with complex condition
if ($x > 0 && $x < 100) {
    print("in range");
}

# If with OR condition
if ($status == 1 || $override) {
    process();
}

# If with NOT condition
if (!$disabled) {
    enable_feature();
}

# Nested if statements
if ($x > 0) {
    if ($x < 10) {
        print("single digit positive");
    }
    if ($x >= 10) {
        print("multi digit positive");
    }
}

# Empty if block
if ($condition) {
}

# If with expression condition
if (($x + $y) > 100) {
    print("sum exceeds 100");
}

# If with function call condition
if (check_status()) {
    proceed();
}

# If with assignment in condition
if ($result = compute()) {
    use_result($result);
}

# Multiple independent if statements
if ($a > 0) {
    handle_positive($a);
}
if ($b < 0) {
    handle_negative($b);
}
if ($c == 0) {
    handle_zero();
}