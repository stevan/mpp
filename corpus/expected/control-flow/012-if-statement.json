[
  {
    "type": "If",
    "condition": {
      "type": "BinaryOp",
      "operator": ">",
      "left": {
        "type": "Variable",
        "name": "$x"
      },
      "right": {
        "type": "Number",
        "value": "10"
      }
    },
    "thenBlock": [
      {
        "type": "Call",
        "name": "print",
        "arguments": [
          {
            "type": "String",
            "value": "\"big\""
          }
        ]
      }
    ],
    "elseIfClauses": []
  },
  {
    "type": "If",
    "condition": {
      "type": "Variable",
      "name": "$valid"
    },
    "thenBlock": [
      {
        "type": "Assignment",
        "left": {
          "type": "Variable",
          "name": "$count"
        },
        "operator": "=",
        "right": {
          "type": "BinaryOp",
          "operator": "+",
          "left": {
            "type": "Variable",
            "name": "$count"
          },
          "right": {
            "type": "Number",
            "value": "1"
          }
        }
      },
      {
        "type": "Call",
        "name": "print",
        "arguments": [
          {
            "type": "Variable",
            "name": "$count"
          }
        ]
      },
      {
        "type": "Call",
        "name": "log_message",
        "arguments": [
          {
            "type": "String",
            "value": "\"incremented\""
          }
        ]
      }
    ],
    "elseIfClauses": []
  },
  {
    "type": "If",
    "condition": {
      "type": "BinaryOp",
      "operator": "&&",
      "left": {
        "type": "BinaryOp",
        "operator": ">",
        "left": {
          "type": "Variable",
          "name": "$x"
        },
        "right": {
          "type": "Number",
          "value": "0"
        }
      },
      "right": {
        "type": "BinaryOp",
        "operator": "<",
        "left": {
          "type": "Variable",
          "name": "$x"
        },
        "right": {
          "type": "Number",
          "value": "100"
        }
      }
    },
    "thenBlock": [
      {
        "type": "Call",
        "name": "print",
        "arguments": [
          {
            "type": "String",
            "value": "\"in range\""
          }
        ]
      }
    ],
    "elseIfClauses": []
  },
  {
    "type": "If",
    "condition": {
      "type": "BinaryOp",
      "operator": "||",
      "left": {
        "type": "BinaryOp",
        "operator": "==",
        "left": {
          "type": "Variable",
          "name": "$status"
        },
        "right": {
          "type": "Number",
          "value": "1"
        }
      },
      "right": {
        "type": "Variable",
        "name": "$override"
      }
    },
    "thenBlock": [
      {
        "type": "Call",
        "name": "process",
        "arguments": []
      }
    ],
    "elseIfClauses": []
  },
  {
    "type": "If",
    "condition": {
      "type": "UnaryOp",
      "operator": "!",
      "operand": {
        "type": "Variable",
        "name": "$disabled"
      }
    },
    "thenBlock": [
      {
        "type": "Call",
        "name": "enable_feature",
        "arguments": []
      }
    ],
    "elseIfClauses": []
  },
  {
    "type": "If",
    "condition": {
      "type": "BinaryOp",
      "operator": ">",
      "left": {
        "type": "Variable",
        "name": "$x"
      },
      "right": {
        "type": "Number",
        "value": "0"
      }
    },
    "thenBlock": [
      {
        "type": "If",
        "condition": {
          "type": "BinaryOp",
          "operator": "<",
          "left": {
            "type": "Variable",
            "name": "$x"
          },
          "right": {
            "type": "Number",
            "value": "10"
          }
        },
        "thenBlock": [
          {
            "type": "Call",
            "name": "print",
            "arguments": [
              {
                "type": "String",
                "value": "\"single digit positive\""
              }
            ]
          }
        ],
        "elseIfClauses": []
      },
      {
        "type": "If",
        "condition": {
          "type": "BinaryOp",
          "operator": ">=",
          "left": {
            "type": "Variable",
            "name": "$x"
          },
          "right": {
            "type": "Number",
            "value": "10"
          }
        },
        "thenBlock": [
          {
            "type": "Call",
            "name": "print",
            "arguments": [
              {
                "type": "String",
                "value": "\"multi digit positive\""
              }
            ]
          }
        ],
        "elseIfClauses": []
      }
    ],
    "elseIfClauses": []
  },
  {
    "type": "If",
    "condition": {
      "type": "Variable",
      "name": "$condition"
    },
    "thenBlock": [],
    "elseIfClauses": []
  },
  {
    "type": "If",
    "condition": {
      "type": "BinaryOp",
      "operator": ">",
      "left": {
        "type": "BinaryOp",
        "operator": "+",
        "left": {
          "type": "Variable",
          "name": "$x"
        },
        "right": {
          "type": "Variable",
          "name": "$y"
        }
      },
      "right": {
        "type": "Number",
        "value": "100"
      }
    },
    "thenBlock": [
      {
        "type": "Call",
        "name": "print",
        "arguments": [
          {
            "type": "String",
            "value": "\"sum exceeds 100\""
          }
        ]
      }
    ],
    "elseIfClauses": []
  },
  {
    "type": "If",
    "condition": {
      "type": "Call",
      "name": "check_status",
      "arguments": []
    },
    "thenBlock": [
      {
        "type": "Call",
        "name": "proceed",
        "arguments": []
      }
    ],
    "elseIfClauses": []
  },
  {
    "type": "If",
    "condition": {
      "type": "BinaryOp",
      "operator": "=",
      "left": {
        "type": "Variable",
        "name": "$result"
      },
      "right": {
        "type": "Call",
        "name": "compute",
        "arguments": []
      }
    },
    "thenBlock": [
      {
        "type": "Call",
        "name": "use_result",
        "arguments": [
          {
            "type": "Variable",
            "name": "$result"
          }
        ]
      }
    ],
    "elseIfClauses": []
  },
  {
    "type": "If",
    "condition": {
      "type": "BinaryOp",
      "operator": ">",
      "left": {
        "type": "Variable",
        "name": "$a"
      },
      "right": {
        "type": "Number",
        "value": "0"
      }
    },
    "thenBlock": [
      {
        "type": "Call",
        "name": "handle_positive",
        "arguments": [
          {
            "type": "Variable",
            "name": "$a"
          }
        ]
      }
    ],
    "elseIfClauses": []
  },
  {
    "type": "If",
    "condition": {
      "type": "BinaryOp",
      "operator": "<",
      "left": {
        "type": "Variable",
        "name": "$b"
      },
      "right": {
        "type": "Number",
        "value": "0"
      }
    },
    "thenBlock": [
      {
        "type": "Call",
        "name": "handle_negative",
        "arguments": [
          {
            "type": "Variable",
            "name": "$b"
          }
        ]
      }
    ],
    "elseIfClauses": []
  },
  {
    "type": "If",
    "condition": {
      "type": "BinaryOp",
      "operator": "==",
      "left": {
        "type": "Variable",
        "name": "$c"
      },
      "right": {
        "type": "Number",
        "value": "0"
      }
    },
    "thenBlock": [
      {
        "type": "Call",
        "name": "handle_zero",
        "arguments": []
      }
    ],
    "elseIfClauses": []
  }
]